### PR TITLE
remove dummy init calls for submachines

### PR DESCRIPTION
**Pull request by leonardoalt**

### CODE REVIEW

File: pipeline/src/test_util.rs [hunk 1]
```diff
@@ -125,7 +125,7 @@ pub fn run_pilcom_with_backend_variant(
             for entry in fs::read_dir(out_dir).unwrap() {
                 let entry = entry.unwrap();
                 let path = entry.path();
-                if path.is_dir() {
+                if path.is_dir() && fs::read_dir(&path).unwrap().count() > 0 {
```
Comment (georgwiese) [inline pipeline/src/test_util.rs:128]:
```suggestion
                // In the composite backend, the prover can choose to skip some machines, which happens
                // if they are empty.
                if path.is_dir() && fs::read_dir(&path).unwrap().count() > 0 {
```

File: riscv/src/large_field/runtime.rs [hunk 2]
```diff
@@ -430,41 +342,6 @@ impl Runtime {
                 ),
             ],
             32,
-            // calling ec_double for machine initialization.
```
Comment (georgwiese) [inline riscv/src/large_field/runtime.rs:433]: Haha, so good to see this gone

**Review by georgwiese — COMMENTED** [review]
Awesome! 🚀

File: pipeline/src/test_util.rs (hunk 1 above)
Comment (leonardoalt) [reply pipeline/src/test_util.rs:130]: done

**Review by georgwiese — APPROVED** [review]

**Pull request merged**